modules/req.js

```diff
@@ -1,49 +1,129 @@
   return new Promise((resolve, reject) => {
-    let url = cliConfs.getApiUrl() + path;
-
-    request.get({
-      headers: {
-        "x-auth-token": config.token
-      },
-      url: url,
-      json: true,
-    }, function optionalCallback(err, httpResponse, body) {
-      if (err || httpResponse.statusCode != 200) {
-        reject(body);
-      } else {
-        resolve(body);
-      }
-    });
-  });
+    if (url === null) url = cliConfs.getApiUrl() + path;
+    fetch(url, {
+        headers: {
+          "content-type": "application/json",
+          "x-auth-token": config.token,
+          'User-Agent': 'express'
+        },
+      })
+      .then(response => {
+        if (response.ok) {
+          if (!bool) {
+            response.json().then(function (data) {
+              resolve(data);
+            })
+          } else resolve(true)
+        } else {
+          if (!bool) {
+            response.json().then(function (data) {
+              reject(data);
+            })
+          } else resolve(false)
+        }
+      })
+      .catch(function (err) {
+        console.log(err)
+      })
+  })
 }
 
-function post(path, params, config) {
+function post(path, params, config, url = null) {
   return new Promise((resolve, reject) => {
-    let url = cliConfs.getApiUrl() + path;
+    if (url === null) url = cliConfs.getApiUrl() + path;
+    fetch(url, {
+        method: 'POST',
+        headers: {
+          "content-type": "application/json",
+          "x-auth-token": config.token,
+          'User-Agent': 'express'
+        },
+        body: JSON.stringify(params)
+      })
+      .then(response => {
+        if (response.ok) {
+          response.json().then(function (data) {
+            resolve(data);
+          })
+        } else {
+          response.json().then(function (data) {
+            reject(data);
+          })
+        }
+      })
+      .catch(function (err) {
+        console.log(err)
+      })
+  })
+}
+
 
-    request.post({
-      headers: {
-        "x-auth-token": config.token
-      },
-      url: url,
-      json: true,
-      form: params
-    }, function optionalCallback(err, httpResponse, body) {
-      if (err || httpResponse.statusCode != 200) {
-        reject(body);
-      } else {
-        resolve(body);
-      }
-    });
-  });
+function upload(path, params, config, url = null, Length = 0) {
+  return new Promise((resolve, reject) => {
+    if (url === null) url = cliConfs.getApiUrl() + path;
+    fetch(url, {
+        method: 'POST',
+        headers: {
+	  "contentType": "application/json",
+          "x-auth-token": config.token,
+          'User-Agent': 'express',
+	  "Content-Length": Length 
+        },
+        body: params
+      })
+      .then(response => {
+        if (response.ok) {
+          response.text().then(function (data) {
+            resolve(data);
+          })
+        } else {
+          response.text().then(function (data) {
```

Comment:
it's .text() only when uploading?

Reply:
I would expect text body response in upload form-data content.
can be changed to json if you confirm the backend is a json doc.